clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -688,6 +699,26 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
             if (shard->prev_entry_.marker.type != TRACE_TYPE_MARKER ||
                 shard->prev_entry_.marker.marker_type != TRACE_MARKER_TYPE_RSEQ_ABORT) {
                 shard->retaddr_stack_.push(0);
+                // When a signal arrives after a branch, function markers will appear in
+                // the trace when the corresponding syscall xfer happens. When a
+                // kernel xfer appears in the beginning of a trace before any
+                // instructions are recorded, we assume the signal arrives after a
+                // branch to avoid false positives.
+                shard->delayed_function_marker_stack_.push(
```

Comment:
I don't think we need the concept of "delayed" at all: we can use last_instr_in_cur_context to always have the right pre-signal instr, and if that's clear then we'll know the trace started mid-signal.

(If we did want to add another stack that tracks signals: best to add a new field to signal_stack_ instead of making a separate stack, I would think.)

Reply:
I've updated the code to use last_instr_in_cur_context.